### PR TITLE
[Feature] Added Message "SignIn" Menu Action

### PR DESCRIPTION
**Pull request by marcomaddy1982**
### Causes

To digital sign a document (PDF) it is necessary add the "SignIn" action on the message menu actions as entry point in the UI.

### Solutions

- Added the UIMenuItem and the Action associated
- Logic to control if the action should be available or not.

### Dependencies

wire-ios-data-model: https://github.com/wireapp/wire-ios-data-model/pull/925
JIRA: https://wearezeta.atlassian.net/browse/ZIOS-12763

### CODE REVIEW

**Comment by typfel** [comment]
Let's change the  destination to the feature branch?

File: Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift [hunk 1]
```diff
@@ -20,6 +20,14 @@ import Foundation
 
 extension ZMConversationMessage {
 
+    /// Whether the message can be digital signed in.
+    var canDigitalSignedIn: Bool {
```
Comment (typfel) [inline Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift]: Maybe re-phrase as `canBeDigitallySigned`? I read `SignedIn` as if we are authenticating (signing in)

File: Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift [hunk 2]
```diff
@@ -20,6 +20,14 @@ import Foundation
...
+        guard sender?.phoneNumber != nil else {
```
Comment (typfel) [inline Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift]: It's the self user who should have a phone number, not the sender of the message.

Comment (marcomaddy1982) [reply Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift]: I know but I thought that was the user. No problem I change it.

File: Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift [hunk 3]
```diff
@@ -20,6 +20,15 @@ import Foundation
 
 extension ZMConversationMessage {
 
+    /// Whether the message can be digitally signed in.
+    var canBeDigitallySigned: Bool {
+        
+        guard ZMUser.selfUser()?.phoneNumber != nil else {
```
Comment (typfel) [inline Wire-iOS/Sources/Helpers/syncengine/ZMMessage+Actions.swift]:
```suggestion
        guard SelfUser.current.phoneNumber != nil else {
```

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageActionController.swift [hunk 4]
```diff
@@ -147,6 +150,10 @@ final class ConversationMessageActionController: NSObject {
         responder?.perform(action: action, for: message, view: view)
     }
 
+    @objc func digitalSignInMessage() {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageActionController.swift]:
```suggestion
    @objc func digitallySignMessage() {
```

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift [hunk 5]
```diff
@@ -23,6 +23,7 @@ enum MessageAction {
     `delete`,
     present,
     save,
+    digitalSignIn,
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift]:
```suggestion
    digitallySign,
```

**Pull request merged**